evm/chains/sharding/mainchain_handler/mainchain_handler.py

```diff
@@ -0,0 +1,102 @@
+import time
+
+import rlp
+
+from eth_utils import (
+    to_checksum_address,
+)
+
+from evm.chains.sharding.mainchain_handler.config import (
+    GASPRICE,
+    TX_GAS,
+)
+
+
+class MainchainHandler:
+
+    def __init__(self, web3_instance, use_eth_tester=False):
+        self._use_eth_tester = use_eth_tester
+        self._w3 = web3_instance
+        assert self._w3.isConnected()
+
+    # RPC related
+
+    def get_block_by_number(self, block_number):
+        return self._w3.eth.getBlock(block_number)
+
+    def get_block_number(self):
+        return self._w3.eth.blockNumber
+
+    def get_code(self, address):
+        address = to_checksum_address(address)
+        return self._w3.eth.getCode(address)
+
+    def get_nonce(self, address):
+        address = to_checksum_address(address)
+        return self._w3.eth.getTransactionCount(address)
+
+    def import_privkey(self, privkey, passphrase):
```

Comment:
Assuming `import_privkey` is an API that is only used in testing, I think it (and any others that are **only** for facilitating testing) should be removed.  We can still add a utiity to do these things elsewhere.

Reply:
You're right. I am moving it to the tests.